cli/program_state.rs

```diff
@@ -115,89 +119,69 @@ impl ProgramState {
 
     if self.flags.no_check {
-      debug!("Transpiling root: {}", module_specifier);
-      // TODO(kitsonk) note that self.permissions != permissions, which is
-      // something that should be handled better in the future.
-      let handler =
-        Rc::new(RefCell::new(FetchHandler::new(self, permissions.clone())?));
-      let mut builder = GraphBuilder2::new(handler, maybe_import_map);
-      builder.insert(&module_specifier).await?;
-      let mut graph = builder.get_graph(&self.lockfile)?;
-
       let (stats, maybe_ignored_options) =
         graph.transpile(TranspileOptions {
-          debug: self.flags.log_level == Some(log::Level::Debug),
-          maybe_config_path: self.flags.config_path.clone(),
+          debug,
+          maybe_config_path,
+          reload: self.flags.reload,
         })?;
-
+      debug!("{}", stats);
       if let Some(ignored_options) = maybe_ignored_options {
         eprintln!("{}", ignored_options);
       }
-
-      debug!("{}", stats);
     } else {
-      let mut module_graph_loader = ModuleGraphLoader::new(
-        self.file_fetcher.clone(),
-        maybe_import_map,
-        permissions.clone(),
-        is_dyn_import,
-        false,
-      );
-      module_graph_loader
-        .add_to_graph(&module_specifier, maybe_referrer)
-        .await?;
-      let module_graph = module_graph_loader.get_graph();
-
-      let out = self
-        .file_fetcher
-        .fetch_cached_source_file(&module_specifier, permissions.clone())
-        .expect("Source file not found");
-
-      let module_graph_files = module_graph.values().collect::<Vec<_>>();
-      // Check integrity of every file in module graph
-      if let Some(ref lockfile) = self.lockfile {
-        let mut g = lockfile.lock().unwrap();
-
-        for graph_file in &module_graph_files {
-          let check_passed =
-            g.check_or_insert(&graph_file.url, &graph_file.source_code);
-
-          if !check_passed {
-            eprintln!(
-              "Subresource integrity check failed --lock={}\n{}",
-              g.filename.display(),
-              graph_file.url
-            );
-            std::process::exit(10);
+      let lib = match target_lib {
+        TargetLib::Main => {
+          if self.flags.unstable {
+            TypeLib::UnstableDenoWindow
+          } else {
+            TypeLib::DenoWindow
           }
         }
-      }
-
-      // Check if we need to compile files.
-      let should_compile = needs_compilation(
-        self.ts_compiler.compile_js,
-        out.media_type,
-        &module_graph_files,
-      );
-      let allow_js = should_allow_js(&module_graph_files);
+        TargetLib::Worker => {
+          if self.flags.unstable {
+            TypeLib::UnstableDenoWorker
+          } else {
+            TypeLib::DenoWorker
+          }
+        }
+      };
+      let (stats, diagnostics, maybe_ignored_options) =
+        graph.check(CheckOptions {
+          debug,
+          emit: true,
```

Comment:
Just to be sure `emit: true` is temporary and we'll remove emit from TSC in the future, right?

Reply:
If we are happy with the output of `tsc` and `isolatedModules`.